File: Programmers/2112/가장 큰 수.py
def solution(numbers):
    new_numbers = []

    max_num = max(numbers)
    size = len(str(max_num)) + 1  # size +1 해줘야함

    for num in numbers:
        # 튜플로
        new_numbers.append((num, str(num) * size))

    new_numbers.sort(reverse=True, key=lambda x: x[1])
    # 문자열은 앞에서부터 비교하므로
    answer = ''

    for nt in new_numbers:
        answer += str(nt[0])

    if sum(numbers) == 0:  # 모두 0인 경우, 예외처리
        answer = '0'

    return answer


print(solution([6, 10, 2]))
print(solution([3, 30, 34, 5, 9]))

File: Programmers/2203/가장 큰 수.js
function solution(numbers) {
  let answer = "";

  newNumbers = [];
  numbers.forEach((num) => {
    let newNum = String(num);
    for (let i = 0; i < 2; i++) {
      newNum += String(num);
    }
    newNumbers.push([newNum, num]);
  });

  newNumbers.sort().reverse();
  console.log(newNumbers);

  for (const [_, num] of newNumbers) {
    answer += num;
  }

  if (numbers.reduce((sum, curr) => sum + curr) === 0) {
    answer = "0";
  }
  return answer;
}

//console.log(solution([6, 10, 2]));
console.log(solution([3, 30, 34, 5, 9]));
